### PR TITLE
Update `Store.ifLet` docs

### PR DESCRIPTION
**Pull request by stephencelis**
#497 brought in a good clarification, but re-reading through the original docs and things felt a bit muddy. Here's an attempt at a quick rewrite.

### CODE REVIEW

File: Sources/ComposableArchitecture/UIKit/IfLetUIKit.swift [hunk 1]
```diff
@@ -75,16 +75,4 @@ extension Store {
       unwrapCancellable.cancel()
     }
   }
-
-  /// An overload of `ifLet(then:else:)` for the times that you do not want to handle the `else`
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/UIKit/IfLetUIKit.swift:79]: This overload was originally added to support trailing closure syntax (`.ifLet { ... }`), but since Swift 5.3, we get this automatically because of [SE-0279](https://github.com/apple/swift-evolution/blob/80602141f65b3fc8a98283e1e8dc4fe1b51312bd/proposals/0279-multiple-trailing-closures.md).

(This'll probably break CI, but just wanted to discuss. Can also bring this back.)

File: Sources/ComposableArchitecture/SwiftUI/TextState.swift [hunk 2]
```diff
@@ -41,9 +41,9 @@ import SwiftUI
 /// In the future, should `SwiftUI.Text` and `SwiftUI.LocalizedStringKey` reliably conform to
 /// `Equatable`, `TextState` may be deprecated.
 ///
-/// - Note: `TextState` does not support _all_ `LocalizedStringKey` permutations at this time, in
-///   particular, for example interpolated `SwiftUI.Image`s. `TextState` also uses reflection to
-///   determine `LocalizedStringKey` equatability, so look out for edge cases.
+/// - Note: `TextState` does not support _all_ `LocalizedStringKey` permutations at this time
+///   (interpolated `SwiftUI.Image`s, for example. `TextState` also uses reflection to determine
+///   `LocalizedStringKey` equatability, so be mindful of edge cases.
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/SwiftUI/TextState.swift:46]: (Unrelated docs cleanup.)

**Review by mbrandonw — APPROVED** [review]
